File: LeetCode_Solutions/1694-make-sum-divisible-by-p/make-sum-divisible-by-p.py
class Solution:
    def minSubarray(self, nums: List[int], p: int) -> int:
        tsum=sum(nums)
        rem=tsum%p
        if rem==0:
            return 0
        mp={}
        mp[0]=-1
        cur=0
        ans=len(nums)
        for i in range(len(nums)):
            cur=(cur+nums[i])%p
            need=(cur-rem+p)%p
            if need in mp:
                ans=min(ans,i-mp[need])
            mp[cur]=i
        return -1 if ans==len(nums)  else ans

File: LeetCode_Solutions/1694-make-sum-divisible-by-p/make-sum-divisible-by-p.java
class Solution {
    public int minSubarray(int[] nums, int p) {
        //int sum=Arrays.stream(nums).sum();
        int sum=0;
        for(int x:nums){
            sum=(sum+x)%p;
        }
        int rem=sum%p;
        if(rem==0){
            return 0;
        }
        int ans=nums.length;
        HashMap<Integer,Integer>hm=new HashMap<>();
        hm.put(0,-1);
        int cur=0;
        for(int i=0;i<nums.length;i++){
            cur=(cur+nums[i])%p;
            int need=(cur-rem+p)%p;
            if(hm.containsKey(need)){
                ans=Math.min(ans,i-hm.get(need));

            }
            hm.put(cur,i);
        }
        //System.out.println(hm);
        return ans==nums.length?-1:ans;
    }
}